hs_core/models.py

```diff
@@ -693,21 +703,22 @@ def create(cls, **kwargs):
             metadata_obj = kwargs['content_object']
             # get matching resource
             resource = BaseResource.objects.filter(object_id=metadata_obj.id).first()
+            publisher_CUAHSI = "Consortium of Universities for the Advancement of Hydrologic Science, Inc. (CUAHSI)"
+            if not resource.raccess.published:
+                raise ValidationError("Publisher element can't be created for a resource that is not yet published.")
 
-            if not resource.raccess.public and resource.raccess.published:
-                raise ValidationError("Publisher element can't be created for a resource that is not yet shared "
-                                      "nor published.")
-
-            if kwargs['name'].lower() == 'hydroshare':
-                if not resource.files.all():
-                    raise ValidationError("Hydroshare can't be the publisher for a resource that has no content "
+            # if the resource has content files, set CUAHSI as the publisher - ignore passed kwargs
+            if resource.files.all():
+                kwargs['name'] = publisher_CUAHSI
+                kwargs['url'] = 'https://www.cuahsi.org'
```

Comment:
@pkdash I think it is better to check whether passed in name and url are correct, i.e., name is equal to publisher_CUAHSI and url is https://www.cuahsi.org, rather than just ignore the passed in name and url and set them correctly. This can validate the calling function knows what it is supposed to do before publishing a resource. If the passed in name and url are not correct, a ValidationError should be raised to inform the caller to raise awareness.